cmd/prometheus/config.go

```diff
@@ -242,10 +242,13 @@ func init() {
 
 func parse(args []string) error {
 	err := cfg.fs.Parse(args)
-	if err != nil {
+	if err != nil || len(cfg.fs.Args()) != 0 {
 		if err != flag.ErrHelp {
 			log.Errorf("Invalid command line arguments. Help: %s -h", os.Args[0])
 		}
+		if err == nil {
+			err = fmt.Errorf("Non flag argument on command line. \"%s\"", cfg.fs.Args()[0])
```

Comment:
- Nit: "Non flag" -> "Non-flag" when used as a compound adjective is nicer.
- Instead of `\"%s\"`, do `%q`.
- "line." -> "line:"